### PR TITLE
Deprecate lib/embulk/version.rb `Embulk::VERSION`, and add `org.embulk.EmbulkVersion` with auto-generated embulk.properties.

### PR DESCRIPTION
**Pull request by dmikurube**
@muga Controlling the version number in one place from Gradle, and in Java world. What do you think?

### CODE REVIEW

File: embulk.gemspec [hunk 1]
```diff
@@ -1,9 +1,6 @@
-$LOAD_PATH.push File.expand_path("../lib", __FILE__)
-require 'embulk/version'
-
 Gem::Specification.new do |gem|
   gem.name          = "embulk"
-  gem.version       = Embulk::VERSION
+  gem.version       = "0.8.18"
```
Comment (dmikurube) [inline embulk.gemspec]: We still need to automate this!

File: lib/embulk/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module Embulk
-  VERSION = '0.8.18'
+  VERSION = org.embulk.EmbulkVersion::VERSION
```
Comment (dmikurube) [inline lib/embulk/version.rb]: We may need to keep this until v2 for forward compatibility of `selfupdate`.

See also: https://github.com/embulk/embulk/blob/4c4d114e84173af81649fc60eab555253baa30a0/lib/embulk/command/embulk_selfupdate.rb#L64-L69

**Comment by dmikurube** [comment]
@muga Changed the approach. As we plan to end `embulk.gem`, `lib/embulk/version.rb` is finally not necessary. Put deprecation message there, and will remove in v2.

**Comment by dmikurube** [comment]
@muga Renamed `version.properties` to `embulk.properties` so that it can contain more information than just its version.

I think it's about time to start merging these un-JRuby-ing PRs. What do you think?

**Comment by muga** [comment]
Sorry for the delay. I will review this PR tomorrow.

**Comment by dmikurube** [comment]
`embulk.gemspec` shouldn't have been revised like that. Keeping it almost as-is. It will be removed at the time of v2.

**Comment by muga** [comment]
I'm checking #562, #496, #563 as well.

I also confirmed that basic embulk commands works fine on my laptop but, to be honest, I don't know how to test such kind of these PRs.

**Comment by dmikurube** [comment]
@muga Other PRs (e.g. #563) may need to be updated/modified due to this #560. We may need to look into Issues, but I think we can leave other PRs at the moment.

**Comment by hiroyuki-sato** [comment]
@dmikurube @muga
In shoft. LGTM.

I compare the latest version and java-embulk-version(8f17cedbe3e55f98875850735871ddd34cc3d5b9) branch.

The following tests are completely the same result. 

This PR does not fix #496 yet. Is this correct?

## migrate check

the latest version (v0.8.18)

```
embulk migrate .
2017-04-05 16:30:59.871 +0900: Embulk v0.8.18
Detected Java plugin for Embulk 0.8.13...
  Modified /gradle/wrapper/gradle-wrapper.properties
  Modified /gradle/wrapper/gradle-wrapper.jar
  Modified /build.gradle
Done. Please check modifieid files.
```

java-embulk-version(8f17cedbe3e55f98875850735871ddd34cc3d5b9)

```
2017-04-05 16:32:14.256 +0900: Embulk v0.8.18
Detected Java plugin for Embulk 0.8.13...
  Modified /gradle/wrapper/gradle-wrapper.properties
  Modified /gradle/wrapper/gradle-wrapper.jar
  Modified /build.gradle
```

## diff check.

Both migrations results complete the same. 

the latest version (v0.8.18)


```diff
diff --git a/build.gradle b/build.gradle
index 5d8b9d5..d894c6f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -20,14 +20,14 @@ sourceCompatibility = 1.7
 targetCompatibility = 1.7
 
 dependencies {
-    compile  "org.embulk:embulk-core:0.8.13"
-    provided "org.embulk:embulk-core:0.8.13"
+    compile  "org.embulk:embulk-core:0.8.18"
+    provided "org.embulk:embulk-core:0.8.18"
 
     compile 'com.jayway.jsonpath:json-path:2.2.0'
 
     testCompile "junit:junit:4.+"
-    testCompile  "org.embulk:embulk-core:0.8.13:tests"
-    testCompile  "org.embulk:embulk-standards:0.8.13"
+    testCompile  "org.embulk:embulk-core:0.8.18:tests"
+    testCompile  "org.embulk:embulk-standards:0.8.18"
 }
 
 findbugs {
diff --git a/gradle/wrapper/gradle-wrapper.jar b/gradle/wrapper/gradle-wrapper.jar
index 13372ae..deedc7f 100644
Binary files a/gradle/wrapper/gradle-wrapper.jar and b/gradle/wrapper/gradle-wrapper.jar differ
diff --git a/gradle/wrapper/gradle-wrapper.properties b/gradle/wrapper/gradle-wrapper.properties
index 989348b..722184f 100644
--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Wed Jan 13 12:41:02 JST 2016
+#Sun Jan 08 00:35:58 PST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.2.1-bin.zip
```

java-embulk-version(8f17cedbe3e55f98875850735871ddd34cc3d5b9)


```diff
diff --git a/build.gradle b/build.gradle
index 5d8b9d5..d894c6f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -20,14 +20,14 @@ sourceCompatibility = 1.7
 targetCompatibility = 1.7
 
 dependencies {
-    compile  "org.embulk:embulk-core:0.8.13"
-    provided "org.embulk:embulk-core:0.8.13"
+    compile  "org.embulk:embulk-core:0.8.18"
+    provided "org.embulk:embulk-core:0.8.18"
 
     compile 'com.jayway.jsonpath:json-path:2.2.0'
 
     testCompile "junit:junit:4.+"
-    testCompile  "org.embulk:embulk-core:0.8.13:tests"
-    testCompile  "org.embulk:embulk-standards:0.8.13"
+    testCompile  "org.embulk:embulk-core:0.8.18:tests"
+    testCompile  "org.embulk:embulk-standards:0.8.18"
 }
 
 findbugs {
diff --git a/gradle/wrapper/gradle-wrapper.jar b/gradle/wrapper/gradle-wrapper.jar
index 13372ae..deedc7f 100644
Binary files a/gradle/wrapper/gradle-wrapper.jar and b/gradle/wrapper/gradle-wrapper.jar differ
diff --git a/gradle/wrapper/gradle-wrapper.properties b/gradle/wrapper/gradle-wrapper.properties
index 989348b..722184f 100644
--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Wed Jan 13 12:41:02 JST 2016
+#Sun Jan 08 00:35:58 PST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.2.1-bin.zip
```

## embulk version compare

Both migration results complete the same. 

the latest version (v0.8.18)

```
embulk version
2017-04-05 16:37:36.931 +0900: Embulk v0.8.18
Embulk v0.8.18
```

java-embulk-version(8f17cedbe3e55f98875850735871ddd34cc3d5b9)


```
/path/to/embulk/pkg/embulk-0.8.18.jar version
2017-04-05 16:38:16.697 +0900: Embulk v0.8.18
Embulk v0.8.18
```

## embulk selfupdate version output

the latest version (v0.8.18)

```
embulk selfupdate
2017-04-05 16:43:02.654 +0900: Embulk v0.8.18
Checking the latest version...
Already up-to-date. 0.8.18 is the latest version.
```

java-embulk-version(8f17cedbe3e55f98875850735871ddd34cc3d5b9)

```
/path/to/embulk-0.8.18.jar selfupdate
2017-04-05 16:39:03.731 +0900: Embulk v0.8.18
Checking the latest version...
Already up-to-date. 0.8.18 is the latest version.
```

**Comment by dmikurube** [comment]
@hiroyuki-sato Thanks. That's really helpful confirmation!

**Comment by dmikurube** [comment]
@muga Confirmed `Embulk::VERSION` with this PR on CRuby `require`. I'm not sure of the actual usage of CRuby & Embulk, though...

```
$ gem install --local pkg/embulk-0.8.18.gem
Successfully installed embulk-0.8.18
Parsing documentation for embulk-0.8.18
Installing ri documentation for embulk-0.8.18
Done installing documentation for embulk after 1 seconds
1 gem installed

$ cat foo.rb
require 'embulk/version'

p Embulk::VERSION

$ ruby foo.rb
Embulk::VERSION in (J)Ruby is deprecated. Use org.embulk.EmbulkVersion::VERSION instead. If this message is from a plugin, please tell this to the author of the plugin!
"0.8.18"
```

**Comment by muga** [comment]
@dmikurube LGTM. Thank you for confirming. I noticed that I didn't checkout your commits enough.

**Comment by dmikurube** [comment]
@muga Thanks! Merging this PR...

**Pull request merged**